### PR TITLE
feat(web): add redirect toast for links older than 30 days

### PR DESCRIPTION
**Pull request by silkeholmebonnen**
## Issue

[AVO-654](https://linear.app/electricitymaps/issue/AVO-654/handle-links-older-than-30-days)

## Description

This PR adds a toast that pops up if a user has been redirected to the last date, because the url date was older than 30 days.

### Preview

image

### Double check

- [ ] I have tested my parser changes locally with `poetry run test_parser "zone_key"`
- [ ] I have run `pnpx prettier@2 --write .` and `poetry run format` in the top level directory to format my changes.

### CODE REVIEW

**Comment by silkeholmebonnen** [comment]
Is this a replacement for [AVO-565](https://linear.app/electricitymaps/issue/AVO-535/historical-empty-state), or when do we want to show the empty state?

**Comment by tonypls** [comment]
Great! @cadeban cc for vis

After chatting with Alex, we think it would make the most sense to send people to real time if it's expired. Maybe Cady can take over this branch?

**Comment by VIKTORVAV99** [comment]
> Great! @cadeban cc for vis
> 
> 
> 
> After chatting with Alex, we think it would make the most sense to send people to real time if it's expired.

I also agree with this. It will also be more performant as it should be cached.

**Comment by silkeholmebonnen** [comment]
> Great! @cadeban cc for vis
> 
> After chatting with Alex, we think it would make the most sense to send people to real time if it's expired. Maybe Cady can take over this branch?

Are you taking over from here @cadeban? Or should I continue

**Comment by cadeban** [comment]
Looks like you've already made good progress on this, so I think you should keep going. I think we want this to be part of the launch tomorrow, so let's sync if we're not able to merge before EOD. How does that sound?

File: web/src/locales/en.json [hunk 1]
```diff
@@ -247,7 +247,8 @@
       "daily": "30d",
       "monthly": "12mo",
       "yearly": "all"
-    }
+    },
+    "disclaimer": "This link is beyond the available dates. We've redirected you to the latest available date."
```
Comment (tonypls) [inline web/src/locales/en.json:251]:
```suggestion
    "date-unavailable": "Date unavailable; redirecting to live data"
```

File: web/src/features/time/DateRedirectToast.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    if (isRedirectedToLatestDatetime) {
+      reference.current?.publish();
+    }
+    setIsRedirectedToLatestDatetime(false);
+  }, [isRedirectedToLatestDatetime, reference, setIsRedirectedToLatestDatetime]);
+
+  return (
+    <Toast
+      ref={reference}
+      description={t('time-controller.disclaimer')}
```
Comment (tonypls) [inline web/src/features/time/DateRedirectToast.tsx:25]:
```suggestion
      description={t('time-controller.date-unavailable')}
```

File: web/src/locales/en.json (hunk 1 above)
Comment (tonypls) [reply web/src/locales/en.json:251]: Mostly to shorten the toast 👍

**Comment by silkeholmebonnen** [comment]
My last commit is changing from this:
image
to this:
image

**Comment by VIKTORVAV99** [comment]
> My last commit is changing from this: image to this: image

Love it! 💚

**Pull request merged**